src/formpack/schema/fields.py

```diff
@@ -786,11 +781,23 @@ def format(self, val, lang=UNSPECIFIED_TRANSLATION,
                 "summary": only the summary column
                 "details": only the details column
         """
-        cells = dict.fromkeys(self.value_names, "0")
+        if val is None:
+            # If the value is missing, do not imply that any response was
+            # received: fill with empty strings instead of zeros
+            return dict.fromkeys(
+                self.get_value_names(multiple_select=multiple_select), ""
+            )
+
+        cells = dict.fromkeys(
+            self.get_value_names(multiple_select=multiple_select), "0"
+        )
         if multiple_select in ("both", "summary"):
             res = []
             for v in val.split():
-                label = self.choice.options[v]['labels'].get(lang)
+                try:
+                    label = self.choice.options[v]['labels'][lang]
+                except KeyError:
+                    label = None
```

Comment:
Out of interest, is this the recommended method for setting the value of `label` in this case? I know I've often used the `.get()` approach.

Reply:
`.get()` is great, but I also needed to handle the case of `v` not being present in `self.choice.options`. I needed to deal with that here in order to remove the large `try` / `except` block in `format_one_submission()`: https://github.com/kobotoolbox/formpack/pull/235/files#diff-c960a2dd6ed9ad6775416e4a60b6ad56e12958e65ec52e4f6775d1ab9ea580d4L362-L376

Reply:
Ahh yes 👍